giraph-core/src/main/java/org/apache/giraph/writable/kryo/HadoopKryo.java

```diff
@@ -203,15 +277,33 @@ public T execute(Kryo kryo) {
     });
   }
 
+  /**
+   * Returns a kryo which doesn't track objects, hence
+   * no recursive/nested objects should be serialized.
+   * @return Hadoop kryo which doesn't track objects.
+   */
+  public static HadoopKryo getNontrackingKryo() {
```

Comment:
How will someone using kryo know which implementation to use? Can we add some guidance around that?

Reply:
Added more documentation.